api/src/main/java/com/datastrato/gravitino/authorization/SecurableObjectType.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.authorization;
+
+/**
+ * The type of securable object in the Gravitino system. Every type will map one kind of the entity
+ * of the underlying system.
+ */
+public enum SecurableObjectType {
+  /**
+   * A catalog is a collection of metadata from a specific metadata source, like Apache Hive
+   * catalog, Apache Iceberg catalog, JDBC catalog, etc.
+   */
+  CATALOG,
+  /**
+   * A schema is a sub collection of the catalog. The schema can contain filesets, tables, topics,
+   * etc.
+   */
+  SCHEMA,
+  /** A fileset is mapped to a directory on a file system like HDFS, S3, ADLS, GCS, etc. */
+  FILESET,
+  /** A table is mapped the table of relational data sources like Apache Hive, MySQL, etc. */
+  TABLE,
+  /**
+   * A topic is mapped the topic of messaging data sources like Apache Kafka, Apache Pulsar, etc.
+   */
+  TOPIC,
+  /**
+   * A metalake is a concept of tenant. It means an organization. A metalake contains many data
+   * sources.
+   */
+  METALAKE,
+  /**
+   * A role is used for access control. A role will indicate that it has some privileges to access
+   * some securable objects.
+   */
+  ROLE
```

Comment:
Why do we need a ROLE securable object?

Reply:
We will create a special role for one entity.
One creator of one entity should have the privilege to grant the owner role to others.

Reply:
Can you give an example? If it is user A who create a table "foo", so the privilege this "grant role", the securable object is table "foo", grant to user A, am I right? the securable object is table, not role, right?

Reply:
That's ok, too. I will remove the entity.